File: tcpcom/PC/Java/Examples/PearlServer.java
// PearlServer.java

import ch.aplu.jgamegrid.*;
import ch.aplu.tcpcom.*;

public class PearlServer extends GameGrid
  implements TCPServerListener, GGMouseListener
{
  private TCPServer server;
  private boolean gameStarted = false;

  public PearlServer()
  {
    super(8, 6, 70, false);
    addStatusBar(30);
    addMouseListener(this, GGMouse.lPress | GGMouse.lRelease);
    PearlShare.initGame(this);
    show();
    int port = 5000;
    server = new TCPServer(port);
    server.addTCPServerListener(this);
  }

  public void onStateChanged(String state, String msg)
  {
    if (state.equals(TCPServer.PORT_IN_USE))
      setStatusText("TCP port occupied. Restart IDE.");
    else if (state.equals(TCPServer.LISTENING))
    {
      if (!PearlShare.isOver && !gameStarted)
        setStatusText("Waiting for a partner to play");
      if (gameStarted)
      {
        setStatusText("Partner lost.");
        PearlShare.isMyMove = false;
      }
    }
    else if (state.equals(TCPServer.CONNECTED))
    {
      setStatusText("Client connected. Wait for partner's move!");
      gameStarted = true;
    }

    else if (state.equals(TCPServer.MESSAGE))
      PearlShare.handleMessage(this, msg);
  }

  public boolean mouseEvent(GGMouse mouse)
  {
    if (mouse.getEvent() == GGMouse.lPress)
      PearlShare.handleMousePress(this, server, mouse);
    if (mouse.getEvent() == GGMouse.lRelease)
      PearlShare.handleMouseRelease(this, server, mouse);
    return true;
  }

  public static void main(String[] args)
  {
    new PearlServer();
  }
}

File: tcpcom/PC/TigerJython/Examples/PearlServer.py
# PearlServer.py

from gamegrid import *
from tcpcom import TCPServer
import pearlshare
                                                                          
def onMousePressed(e):
    pearlshare.handleMousePress(e, server)

def onMouseReleased(e):
    pearlshare.handleMouseRelease(e)

def onNotifyExit():
    server.terminate()
    dispose()
    
def onStateChanged(state, msg):
    global gameStarted, partnerLost
    if state == TCPServer.PORT_IN_USE:
        setStatusText("TCP port occupied. Restart IDE.")
    elif state == TCPServer.LISTENING:
        if gameStarted:
            setStatusText("Connection lost. Restart server.")
            pearlshare.isMyMove = False
            server.terminate()
        else:
            setStatusText("Waiting for a partner to play")
    elif state == TCPServer.CONNECTED:
        if gameStarted:
            return
        setStatusText("Client connected. Wait for partner's move!")
        gameStarted = True
    elif state == TCPServer.MESSAGE: 
        pearlshare.handleMessage(msg)    
 
makeGameGrid(8, 6, 70, False, mousePressed = onMousePressed, mouseReleased = onMouseReleased, 
                              notifyExit = onNotifyExit)
addStatusBar(30)
pearlshare.initGame()
show()
gameStarted = False
port = 5000
server = TCPServer(port, stateChanged = onStateChanged)
server.terminate()